src/ir.rs

```diff
@@ -567,7 +567,10 @@ pub enum DomainGoal {
     /// Derefs(T, U) :- Implemented(T: Deref<Target = U>)
     /// ```
     /// In Rust there are also raw pointers which can be deref'd but do not implement Deref.
-    Derefs(Derefs)
+    Derefs(Derefs),
+
+    IsLocalTy(Ty),
```

Comment:
Comment here would be nice. At least something minimal like:

---

True if a type is considered to have been "defined" by the current crate. This is true for a `struct Foo { }` but false for a `extern struct Foo { }`. However, for fundamental types like `Box<T>`, it is true if `T` is local.